### PR TITLE
feat: add working days component (#162)

### PR DESCRIPTION
**Pull request by OussEmaDevCode**
## Description
Added the working days component, connected it to the app via routing and implemented the Required API.

## Related issues and discussion
#162 

## Screenshots, if any
Capture d’écran (217)
Capture d’écran (218)

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by OussEmaDevCode** [comment]
@abhaychawla I closed the other PR and created a new one please have a look, hope this one is good !!

**Comment by punwai** [comment]
Please try having a look at your commit message. There is an error on the issue that it is referring to.

**Comment by OussEmaDevCode** [comment]
what do you mean by "an error on the issue" ?

**Comment by punwai** [comment]
Ah, I meant that the commit message is referring to the wrong issue. Sorry for the bad wording.

**Comment by punwai** [comment]
My mistakes, I saw your previous PR which I commented on. You must have referred to the password preferences PR when you were trying to merge it. This one refers to the right issue.

**Comment by OussEmaDevCode** [comment]
It's OK !!

**Comment by punwai** [comment]
For the issue reference though, try seeing #529. It would be better if the issue was referred as 'Fixes: #162' rather than being in brackets

**Comment by punwai** [comment]
I apologise again for the confusion before 😅

File: src/app/organization/working-days/working-days.component.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          <mat-form-field>
+            <mat-label>Payments due on non working days</mat-label>
+            <mat-select formControlName="repaymentRescheduleType">
+              <mat-option *ngFor="let repaymentRescheduleType of repaymentRescheduleTypeData" [value]="repaymentRescheduleType.id">
+                {{ repaymentRescheduleType.value }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+
+            <mat-checkbox labelPosition="before" formControlName="extendTermForDailyRepayments">
```
Comment (abhaychawla) [inline src/app/organization/working-days/working-days.component.html]:
```suggestion
          <mat-checkbox labelPosition="before" formControlName="extendTermForDailyRepayments">
```
Indent properly, remove extra space.

File: src/app/organization/working-days/working-days.component.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+              Extend the term for loans following a daily repayment schedule
```
Comment (abhaychawla) [inline src/app/organization/working-days/working-days.component.html]:
```suggestion
            Extend the term for loans following a daily repayment schedule
```

File: src/app/organization/working-days/working-days.component.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            </mat-checkbox>
```
Comment (abhaychawla) [inline src/app/organization/working-days/working-days.component.html]:
```suggestion
          </mat-checkbox>
```

File: src/app/organization/working-days/working-days.component.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import {FormGroup, FormBuilder, FormArray, FormControl} from '@angular/forms';
```
Comment (abhaychawla) [inline src/app/organization/working-days/working-days.component.ts]:
```suggestion
import { FormGroup, FormBuilder, FormArray, FormControl } from '@angular/forms';
```

**Comment by OussEmaDevCode** [comment]
@abhaychawla Done !

**Comment by abhaychawla** [comment]
@OussEmaDevCode the PR still shows the merge commit. It should not be there. Also, remove the reference to issue in the brackets from the first line.

**Comment by OussEmaDevCode** [comment]
@abhaychawla I couldn't find a way to remove the merge commit, every time I run 'git rebase -i' the commit is not appearing in the text editor

**Comment by abhaychawla** [comment]
You have to resolve those conflicts and then squash the commits into a single commit.

**Comment by Rishabh106** [comment]
@OussEmaDevCode Go to your Editor, then resolve every merge conflict one by one. Then, you can rebase at last.

**Comment by OussEmaDevCode** [comment]
@abhaychawla I have resolved the merge conflicts, my only problem is that I can't squash the last commit this one ''Merge branch 'master' into working-days-component". I can see it when running "git log --oneline" but when running "git rebase -i HEAD~2" it is not in the message as you can see [here](https://user-images.githubusercontent.com/38397893/71561880-e84ea180-2a7c-11ea-81c3-a2048c576629.png) and [here](https://user-images.githubusercontent.com/38397893/71561883-ed135580-2a7c-11ea-9878-3f4e4b6f3057.png).

**Comment by OussEmaDevCode** [comment]
@abhaychawla I've been trying to get this right for almost an hour now but I unfortunately couldn't. This is my first time doing this and I hope it will be my last time but I am sad to say that I give up. I am abandoning the task. This took way too much time 6 days to be exact and it's draining my energy and your time. So sorry @abhaychawla for deceiving you and I can assure you that I'll keep contributing to this repository whenever I can, and @punwai @Rishabh106 feel free to use the source code in this PR if you want to do this task.

**Comment by punwai** [comment]
@oo k, I t

**Comment by punwai** [comment]
@OussEmaDevCode Please keep working on this. You are doing well and you will get better and better 😁👍

**Comment by punwai** [comment]
You are nearly done

**Comment by punwai** [comment]
Can you completely resolve the merge??
Once you are able to do that run:
git fetch upstream master
Then:
git rebase upstream/master

Resolve the merge conflicts, then:
git add . 
Then:
git rebase --continue

Force push this. 
git push origin -f working-days-component

**Comment by punwai** [comment]
Keep it up ;)

**Comment by punwai** [comment]
I won't claim this yet as I believe that you could still do this.

**Comment by Rishabh106** [comment]
Don't lose hope, everything will be fine soon @OussEmaDevCode

**Comment by OussEmaDevCode** [comment]
@abhaychawla Fixed the merge conflicts, did the required changes, It should be good to go !

**Comment by punwai** [comment]
@OussEmaDevCode Legend

**Comment by Rishabh106** [comment]
Well done @OussEmaDevCode

**Pull request merged**